libs/providers/config-cat-web/src/lib/config-cat-web-provider.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  EvaluationContext,
+  FlagNotFoundError,
+  JsonValue,
+  OpenFeatureEventEmitter,
+  Paradigm,
+  ParseError,
+  Provider,
+  ProviderEvents,
+  ProviderNotReadyError,
+  ResolutionDetails,
+} from '@openfeature/web-sdk';
+import { TypeMismatchError } from '@openfeature/core';
+import {
+  isType,
+  PrimitiveType,
+  PrimitiveTypeName,
+  toResolutionDetails,
+  transformContext,
+} from '@openfeature/config-cat-core';
+import {
+  getClient,
+  IConfig,
+  IConfigCatClient,
+  IConfigCatClientSnapshot,
+  OptionsForPollingMode,
+  PollingMode,
+} from 'configcat-js-ssr';
+
+export class ConfigCatWebProvider implements Provider {
+  public readonly events = new OpenFeatureEventEmitter();
+  private readonly _sdkKey: string;
+  private readonly _configCatOptions: OptionsForPollingMode<PollingMode.AutoPoll>;
+  private _client?: IConfigCatClient;
+  private _clientSnapshot?: IConfigCatClientSnapshot;
+
+  public runsOn: Paradigm = 'client';
+
+  public metadata = {
+    name: ConfigCatWebProvider.name,
+  };
+
+  constructor(sdkKey: string, options: OptionsForPollingMode<PollingMode.AutoPoll> = {}) {
+    this._sdkKey = sdkKey;
+    this._configCatOptions = options;
+  }
+
+  public static create(sdkKey: string, options: OptionsForPollingMode<PollingMode.AutoPoll> = {}) {
+    return new ConfigCatWebProvider(sdkKey, options);
+  }
+
+  public async initialize(): Promise<void> {
+    const options = this._configCatOptions;
+    const oldSetupHooks = options.setupHooks;
+
+    options.setupHooks = (hooks) => {
+      oldSetupHooks?.(hooks);
+
+      hooks.on('clientReady', () => {
+        this.events.emit(ProviderEvents.Ready);
+      });
```

Comment:
As I described above, subscribing to `clientReady` is problematic. I suggest the same solution (i.e. moving the emitting of `ProviderEvents.Ready` after `waitForReady`).

```suggestion
```